### ISSUE TITLE
Errors reported when running "dotnet build /t:GenerateCode"

### ISSUE THREAD

**Issue by mengaims** [Azure/autorest.csharp#3359]
I've met some errors when running "dotnet build /t:GenerateCode" using remote branch in tsp-location.yaml, after mitigate with referencing to local repository, it works. But the CI build will be blocked by this error and others could not build on their machines.
Refer to tsp-location file here: https://github.com/mengaims/azure-sdk-for-net/blob/carnegie-test/sdk/contentsafety/Azure.AI.ContentSafety/tsp-location.yaml

The error is:
Copy-Item: [My local sdk repository]\eng\common\scripts\TypeSpec-Project-Sync.ps1:23
Line |
23 | Copy-Item -Path $source -Destination $dest -Recurse -Force
| ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| Cannot find path
| '[My local directory]\sparse-spec\Azure.AI.ContentSafety\specification\cognitiveservices\ContentSafety' because it

0m
| does not exist.

@chunyu3 @m-nash could you help have a check?
CC @heaths @catalinaperalta

**Comment by chunyu3** [comment]
it is the permission issue. it reference to the private repo azure-rest-api-specs-pr, it is no permission to sparse checkout code.

**Comment by m-nash** [comment]
We should not be merging anything in `azure-sdk-for-net` that points at the private spec repo.  It should only go into the private sdk repo.

**Comment by heaths** [comment]
Thanks for confirming. I was afraid of that, but wasn't entirely sure. @mengaims can you get the specs moved to the public repo?

**Comment by mengaims** [comment]
@heaths Yes, the specs has been merged. https://github.com/Azure/azure-rest-api-specs/pull/23840

**Comment by heaths** [comment]
@m-nash should we just close this issue as By Design then, or keep it open to track support -pr if that's even desirable?

**Comment by mengaims** [comment]
This error has been resolved after pulling typespec from public repo. I think we can close it now.

**Comment by m-nash** [comment]
Closing this as we have another issue tracking private repo support in the sync / generate scripts https://github.com/Azure/azure-sdk-tools/issues/5275
